### ISSUE TITLE
When adding list items, and hitting enter on empty item, list should be ended and last empty item should be removed

### ISSUE THREAD

**Issue by Strajk** [HappenApps/Quiver#73]
```
* one
* two
* three
* 

rest rest rest
```

should be

```
* one
* two
* three

rest rest rest
```

![33vrq8ut8v](https://cloud.githubusercontent.com/assets/697301/6764517/3f43ad66-cfb3-11e4-91b0-779dfc6fc82a.gif)

**Comment by ylian** [comment]
Unfortunately I don't think I can fix this. This behavior is implemented by the Markdown mode of the Ace editor. I don't know Ace's source code well enough to change this behavior.

**Comment by ylian** [comment]
Reopen this. I realized some online Markdown editors actually can do this and they also use Ace, so this is definitely doable.

**Comment by ylian** [comment]
Fixed in 2.2b32.

https://dl.dropboxusercontent.com/u/3445731/Quiver%202.2b32.zip

**Comment by ylian** [comment]
This is fixed in Quiver 3.0, released today.
